SonarQube.Common/VerbosityCalculator.cs

```diff
@@ -0,0 +1,109 @@
+        /// </summary>
+        /// <remarks>If no verbosity setting is present, the default verbosity (info) is used</remarks>
+        public static LoggerVerbosity ComputeVerbosity(IAnalysisPropertyProvider properties, ILogger logger)
+        {
+            if (properties == null)
+            {
+                throw new ArgumentNullException("properties");
+            }
+
+            if (logger == null)
+            {
+                throw new ArgumentNullException("logger");
+            }
+
+            string sonarVerboseValue;
+            properties.TryGetValue(SonarProperties.Verbose, out sonarVerboseValue);
+
+            string sonarLogLevelValue;
+            properties.TryGetValue(SonarProperties.LogLevel, out sonarLogLevelValue);
+
+            return ComputeVerbosity(sonarVerboseValue, sonarLogLevelValue, logger);
+        }
+
+        /// <summary>
+        /// Computes verbosity based on the analysis config file settings. 
+        /// </summary>
+        /// <remarks>If no verbosity setting is present, the default verbosity (info) is used</remarks>
+        public static LoggerVerbosity ComputeVerbosity(AnalysisConfig config, ILogger logger)
+        {
+            if (config == null)
+            {
+                throw new ArgumentNullException("config");
+            }
+
+            if (logger == null)
+            {
+                throw new ArgumentNullException("logger");
+            }
+
+            AnalysisSetting sonarVerboseSetting;
+            config.TryGetSetting(SonarProperties.Verbose, out sonarVerboseSetting);
+
+            AnalysisSetting sonarLogLevelSetting;
+            config.TryGetSetting(SonarProperties.LogLevel, out sonarLogLevelSetting);
+
+            return ComputeVerbosity(
+                sonarVerboseSetting == null ? null : sonarVerboseSetting.Value,
+                sonarLogLevelSetting == null ? null : sonarLogLevelSetting.Value,
+                logger);
+        }
+
+        private static LoggerVerbosity ComputeVerbosity(string sonarVerboseValue, string sonarLogValue, ILogger logger)
+        {
+            if (!String.IsNullOrWhiteSpace(sonarVerboseValue))
+            {
+                bool isVerbose;
+                if (bool.TryParse(sonarVerboseValue, out isVerbose))
+                {
+                    LoggerVerbosity verbosity = isVerbose ? LoggerVerbosity.Debug : LoggerVerbosity.Info;
+                    logger.LogInfo("sonar.verbose={0} was specified - setting the log verbosity to {1}", sonarVerboseValue, verbosity);
+                    return verbosity;
+                }
+                else
+                {
+                    logger.LogWarning(Resources.WARN_SonarVerboseNotBool, sonarVerboseValue);
+                }
+            }
+
+            if (!String.IsNullOrWhiteSpace(sonarLogValue) &&
+                sonarLogValue.IndexOf("DEBUG", StringComparison.OrdinalIgnoreCase) > -1) //todo: move to a const
```

Comment:
Well the log level can also be INFO|DEBUG - so it's a flag enum. I made the simplifying assumption that if the value contains "DEBUG" the user wants verbose logging.